go/apps/ctrl/services/build/backend/depot/create_build.go

```diff
@@ -0,0 +1,350 @@
+	cliv1 "github.com/depot/depot-go/proto/depot/cli/v1"
+	ctrlv1 "github.com/unkeyed/unkey/go/gen/proto/ctrl/v1"
+	"github.com/unkeyed/unkey/go/pkg/db"
+)
+
+type authTransport struct {
+	token string
+	base  http.RoundTripper
+}
+
+// CreateBuild orchestrates the container image build process using Depot.
+//
+// Steps:
+//
+//	Get or create Depot project
+//	Register a new build with Depot
+//	Acquire a build machine
+//	Connect to the buildkit instance
+//	Prepare build context and configuration
+//	Execute the build with status logging
+//	Return build metadata
+func (s *Depot) CreateBuild(
+	ctx context.Context,
+	req *connect.Request[ctrlv1.CreateBuildRequest],
+) (*connect.Response[ctrlv1.CreateBuildResponse], error) {
+	if req.Msg.ContextKey == "" {
+		return nil, connect.NewError(connect.CodeInvalidArgument,
+			fmt.Errorf("contextKey is required"))
+	}
+	if req.Msg.UnkeyProjectId == "" {
+		return nil, connect.NewError(connect.CodeInvalidArgument,
+			fmt.Errorf("unkeyProjectID is required"))
+	}
+
+	s.logger.Info("Starting build process",
+		"context_key", req.Msg.ContextKey,
+		"unkey_project_id", req.Msg.UnkeyProjectId)
+
+	s.logger.Info("Getting presigned URL for build context",
+		"context_key", req.Msg.ContextKey,
+		"unkey_project_id", req.Msg.UnkeyProjectId)
+
+	contextURL, err := s.storage.GetPresignedURL(ctx, req.Msg.ContextKey, 15*time.Minute)
+	if err != nil {
+		s.logger.Error("Failed to get presigned URL",
+			"error", err,
+			"context_key", req.Msg.ContextKey,
+			"unkey_project_id", req.Msg.UnkeyProjectId)
+		return nil, connect.NewError(connect.CodeInternal,
+			fmt.Errorf("failed to get presigned URL: %w", err))
+	}
+
+	depotProjectID, err := s.getOrCreateDepotProject(ctx, req.Msg.UnkeyProjectId)
+	if err != nil {
+		s.logger.Error("Failed to get/create depot project",
+			"error", err,
+			"unkey_project_id", req.Msg.UnkeyProjectId)
+		return nil, connect.NewError(connect.CodeInternal,
+			fmt.Errorf("failed to get/create depot project: %w", err))
+	}
+
+	s.logger.Info("Creating depot build",
+		"depot_project_id", depotProjectID,
+		"unkey_project_id", req.Msg.UnkeyProjectId)
+
+	buildResp, err := build.NewBuild(ctx, &cliv1.CreateBuildRequest{
+		ProjectId: depotProjectID,
+	}, s.accessToken)
+	if err != nil {
+		s.logger.Error("Creating depot build failed",
+			"error", err,
+			"depot_project_id", depotProjectID,
+			"unkey_project_id", req.Msg.UnkeyProjectId)
+		return nil, connect.NewError(connect.CodeInternal,
+			fmt.Errorf("failed to create build: %w", err))
+	}
+
+	s.logger.Info("Depot build created",
+		"build_id", buildResp.ID,
+		"depot_project_id", depotProjectID,
+		"unkey_project_id", req.Msg.UnkeyProjectId)
+
+	var buildErr error
+	defer buildResp.Finish(buildErr)
+
+	s.logger.Info("Acquiring build machine",
+		"build_id", buildResp.ID,
+		"platform", "arm64",
+		"unkey_project_id", req.Msg.UnkeyProjectId)
+
+	buildkit, buildErr := machine.Acquire(ctx, buildResp.ID, buildResp.Token, "arm64")
```

Comment:
so what if some builds an image for x86_64?

Reply:
yeah so depot has machines for “amd64” or “arm64” but tbh I think we can deal with it if the issue comes up